specification/integration_context.md

```diff
@@ -0,0 +1,91 @@
+<!-- markdownlint-configure-file 
+{ 
+  "MD013": { 
+    "tables": false 
+  } 
+}
+-->
+
+# Integration Context
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+In order to create a more complete picture of a distributed environment, we
+propose the use of additional headers to pass extra context details between
+Cisco observability components. Specifically, we are concerned with passing
+information between OpenTelemetry-based Splunk distributions and AppDynamics
+agents, in both directions.
+
+# Headers
+
+The following defines several new headers and describes their intended use.
+All headers MUST be treated as optional -- peer services will not always
+generate them. These headers SHOULD be treated as opaque values of type
+string.
+
+* `cisco-ctx-acct-id` - Contains the ID of the AppDynamics account.
+* `cisco-ctx-app-id` - Contains the ID of the AppDynamics application.
+* `cisco-ctx-tier-id` - Contains the ID of the AppDynamics tier.
+* `cisco-ctx-bt-id` - Contains the ID of the AppDynamics business transaction (BT).
+* `cisco-ctx-env` - Contains
+  the [deployment.environment.name](https://opentelemetry.io/docs/specs/semconv/attributes-registry/deployment/)
+  resource value from an OpenTelemetry based component.
+* `cisco-ctx-service` - Contains the [service.name](https://opentelemetry.io/docs/specs/semconv/resource/#service)
+  resource value from an OpenTelemetry based component.
+
+HTTP headers are capable of being multivalued. As such, implementations
+SHOULD use the _last_ value when the above headers contain multiple values.
```

Comment:
I think that the common pattern is actually to use the first value :sweat_smile: 
This is also reflected in (most?) standard libraries. E.g.
- Java `getFirst`: https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/Headers.html
- Go `Get`: https://pkg.go.dev/net/http#Header.Get
- Rust: https://docs.rs/http/latest/src/http/header/map.rs.html#730-756

Some background:
- https://github.com/golang/go/issues/51493#issuecomment-1059588158

Moreover, this header controlled by us so producing multiple values for the same header is already a bug. Long story short, it should never happen. Therefore, it does not matter if we take the first or last as we have no guarantee which value is "better".

Reply:
Using the first value is much easier. Note that implementation in https://github.com/signalfx/splunk-otel-java/pull/2198 also uses first value.

Reply:
That's cool, appreciate that. I'll change it. 👍🏻